unknown error when select from hudi external table

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create hudi resource and hudi external table:
```
create external resource hudi_resource PROPERTIES("type"="hudi", "hive.metastore.uris"="thrift://***.***.***.***:9083");
create external table ex_hudi_parquet_gzip (
 uuid int
,col_int int
,col_date date
,col_string string
,col_boolean boolean
,col_binary varchar(200)
,col_float float
,col_double double
,col_long bigint
,col_decimal decimal(38,18)) 
ENGINE=hudi 
properties ( 
"resource" = "hudi_resource",
"table" = "hudi_parquet_gzip",
"database" = "hudi_db");
```

2. unknown error when select from hudi external table
```
mysql> select count(*) as count_num from ex_hudi_parquet_gzip;
ERROR 1064 (HY000): Unknown error
```

fe.warn.log:
```
2022-08-25 23:55:16,786 WARN (starrocks-mysql-nio-pool-26|1839) [StmtExecutor.execute():494] execute Exception, sql insert into primary_tbl select uuid, col_int, col_date, col_string, col_boolean, col_binary, col_float, col_double, col_long, col_decimal from ex_hudi_oss_parquet_lz4 limit 1
com.starrocks.sql.common.StarRocksPlannerException: null
        at com.starrocks.sql.optimizer.rule.transformation.RemoteScanPartitionPruneRule.transform(RemoteScanPartitionPruneRule.java:70) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.TopDownRewriteTask.doExecute(TopDownRewriteTask.java:53) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.TopDownRewriteOnceTask.execute(TopDownRewriteOnceTask.java:31) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.SeriallyTaskScheduler.executeTasks(SeriallyTaskScheduler.java:42) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.ruleRewriteOnlyOnce(Optimizer.java:376) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.logicalRuleRewrite(Optimizer.java:275) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:91) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.InsertPlanner.plan(InsertPlanner.java:106) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:77) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:40) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:351) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:318) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:433) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:671) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
```

### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 1bee3d4 |
+----------------------+
1 row in set (0.01 sec)
```

```
2022-08-26 16:32:07,727 ERROR (stateChangeExecutor|68) [IcebergConnector.getMetadata():38] Failed to create iceberg metadata on [catalog : iceberg2]
java.lang.NullPointerException: Cannot initialize custom GlobalStateMgr, impl class name is null
        at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:897) ~[iceberg-bundled-guava-0.12.1.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader$CustomCatalogLoader.<init>(CatalogLoader.java:87) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader$CustomCatalogLoader.<init>(CatalogLoader.java:72) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader.custom(CatalogLoader.java:36) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.IcebergUtil.getIcebergCustomCatalog(IcebergUtil.java:84) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.IcebergMetadata.<init>(IcebergMetadata.java:48) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.IcebergConnector.getMetadata(IcebergConnector.java:36) [starrocks-fe.jar:?]
        at com.starrocks.connector.ConnectorMgr.registerConnectorInternal(ConnectorMgr.java:114) [starrocks-fe.jar:?]
        at com.starrocks.connector.ConnectorMgr.createConnector(ConnectorMgr.java:71) [starrocks-fe.jar:?]
        at com.starrocks.server.CatalogMgr.replayCreateCatalog(CatalogMgr.java:144) [starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:880) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1702) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1660) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToLeader(GlobalStateMgr.java:937) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$100(GlobalStateMgr.java:270) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$1.transferToLeader(GlobalStateMgr.java:595) [starrocks-fe.jar:?]
        at com.starrocks.ha.StateChangeExecutor.runOneCycle(StateChangeExecutor.java:86) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
2022-08-26 16:32:07,731 ERROR (stateChangeExecutor|68) [EditLog.loadJournal():933] Operation Type 10051
com.starrocks.common.DdlException: Failed to create connector on [catalog : iceberg2, type : iceberg]
        at com.starrocks.connector.ConnectorMgr.createConnector(ConnectorMgr.java:80) ~[starrocks-fe.jar:?]
        at com.starrocks.server.CatalogMgr.replayCreateCatalog(CatalogMgr.java:144) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:880) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1702) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1660) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToLeader(GlobalStateMgr.java:937) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$100(GlobalStateMgr.java:270) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$1.transferToLeader(GlobalStateMgr.java:595) [starrocks-fe.jar:?]
        at com.starrocks.ha.StateChangeExecutor.runOneCycle(StateChangeExecutor.java:86) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException: Cannot initialize custom GlobalStateMgr, impl class name is null
        at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:897) ~[iceberg-bundled-guava-0.12.1.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader$CustomCatalogLoader.<init>(CatalogLoader.java:87) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader$CustomCatalogLoader.<init>(CatalogLoader.java:72) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.CatalogLoader.custom(CatalogLoader.java:36) ~[starrocks-fe.jar:?]
        at com.starrocks.external.iceberg.IcebergUtil.getIcebergCustomCatalog(IcebergUtil.java:84) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.IcebergMetadata.<init>(IcebergMetadata.java:48) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.IcebergConnector.getMetadata(IcebergConnector.java:36) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.ConnectorMgr.registerConnectorInternal(ConnectorMgr.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.ConnectorMgr.createConnector(ConnectorMgr.java:71) ~[starrocks-fe.jar:?]
        ... 9 more
```

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoteScanPartitionPruneRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoteScanPartitionPruneRule.java
@@ -60,7 +60,7 @@ public class RemoteScanPartitionPruneRule extends TransformationRule {
         Map<ColumnRefOperator, Set<Long>> columnToNullPartitions = Maps.newHashMap();
 
         try {
-            initPartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions);
+            initPartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions, context);
             classifyConjuncts(operator, columnToPartitionValuesMap);
             computePartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions);
         } catch (Exception e) {
@@ -72,7 +72,8 @@ public class RemoteScanPartitionPruneRule extends TransformationRule {
 
     private void initPartitionInfo(LogicalScanOperator operator,
                                    Map<ColumnRefOperator, TreeMap<LiteralExpr, Set<Long>>> columnToPartitionValuesMap,
-                                   Map<ColumnRefOperator, Set<Long>> columnToNullPartitions)
+                                   Map<ColumnRefOperator, Set<Long>> columnToNullPartitions,
+                                   OptimizerContext context)
             throws DdlException, AnalysisException {
         Table table = operator.getTable();
         if (table instanceof HiveMetaStoreTable) {
@@ -91,6 +92,9 @@ public class RemoteScanPartitionPruneRule extends TransformationRule {
             // 2. partitionKeys size = 1
             // 3. key.getKeys() is empty
             Map<PartitionKey, Long> partitionKeys = hiveMetaStoreTable.getPartitionKeys();
+            // record partition keys for query dump
+            context.getDumpInfo().getHMSTable(hiveMetaStoreTable.getResourceName(), hiveMetaStoreTable.getDbName(),
+                    hiveMetaStoreTable.getTableName()).setPartitionKeys(partitionKeys);
             for (Map.Entry<PartitionKey, Long> entry : partitionKeys.entrySet()) {
                 PartitionKey key = entry.getKey();
                 long partitionId = entry.getValue();

